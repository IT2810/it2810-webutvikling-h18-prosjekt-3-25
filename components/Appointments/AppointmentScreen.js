import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Icon } from 'react-native-elements';
 
export default class AppointmentScreen extends React.Component {

	static navigationOptions = {
		title: "Appointment",
	}

	//viser valgt appointment med alle detaljer
	render() {
		return (
			<View style={styles.appointmentContainer}>
				<View style={styles.texts}>
					<View style={styles.headerView}>
						<Icon name="star" size={50}/>
						<Text style={styles.header}>{this.props.navigation.getParam("header")}</Text>
					</View>
					<View style={styles.infoLine}>
						<Icon name="event" size={30}/>
						<Text style={styles.text}>{this.props.navigation.getParam("date")}</Text>
					</View>
					<View style={styles.infoLine}>
						<Icon name="description" size={30}/>
						<Text style={styles.text}>{this.props.navigation.getParam("description")}</Text>
					</View>
				</View>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	appointmentContainer: {
		flex: 1,
	},

  texts: {
    flex: 2,
  },
	
	infoLine: {
    flexDirection: 'row',
    paddingLeft: 10,
    borderBottomWidth: 1,
    borderBottomColor: '#aaa',
  },

  text: {
    fontSize: 18,
    paddingBottom: 15,
    paddingTop: 30,
    paddingLeft: 20,
  },

	headerView: {
    flexDirection: 'row',
    paddingLeft: 10,
		borderBottomWidth: 1,
		borderBottomColor:'#87cefa',
  },
  header: {
  	fontSize: 30,
    paddingBottom: 15,
    paddingTop: 15,
    paddingLeft: 20,
  },
});