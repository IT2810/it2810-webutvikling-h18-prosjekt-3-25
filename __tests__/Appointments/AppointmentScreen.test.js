import React from 'react';
import AppointmentScreen from '../../components/Appointments/AppointmentScreen.js';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const navigation = {getParam: jest.fn(() => {
	return "test appointment"
})}

it("renders correctly", () => {
    const tree = renderer.create(<AppointmentScreen navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
})