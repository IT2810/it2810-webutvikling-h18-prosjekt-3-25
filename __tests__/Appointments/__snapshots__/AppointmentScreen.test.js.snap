// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#87cefa",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingLeft": 10,
        }
      }
    >
      <View
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 50,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 30,
            "paddingBottom": 15,
            "paddingLeft": 20,
            "paddingTop": 15,
          }
        }
      >
        test appointment
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#aaa",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingLeft": 10,
        }
      }
    >
      <View
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "paddingBottom": 15,
            "paddingLeft": 20,
            "paddingTop": 30,
          }
        }
      >
        test appointment
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#aaa",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingLeft": 10,
        }
      }
    >
      <View
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "paddingBottom": 15,
            "paddingLeft": 20,
            "paddingTop": 30,
          }
        }
      >
        test appointment
      </Text>
    </View>
  </View>
</View>
`;
